docs/zkapps/advanced-snarkyjs/actions-and-reducer.mdx

```diff
@@ -99,9 +99,63 @@ this.reducer.getActions({ fromActionsHash?: Field, endActionsHash?: Field }): A[
 
 Right now, `getActions` is available for [testing with `LocalBlockchain`](/zkapps/how-to-test-a-zkapp).
 
+### Fetching Actions from an Archive Node
+
+Since Mina nodes do not store historical network information and actions are not kept on-chain, emitted actions are only preserved in the Archive Node. Consequently, a zkApp must fetch its previously emitted actions on the Mina network from an Archive Node.
+
+As the Archive Node is essentially a database, it does not offer a user-friendly API that SnarkyJS can readily utilize to fetch actions. To address this issue, a GraphQL server was developed to retrieve network information such as actions and actions from the Archive Node. SnarkyJS will use this GraphQL server to fetch actions from the Archive Node.
+
+To configure SnarkyJS to use an Archive Node to fetch actions, use the `archive` property in the `Mina.Network` function. The `archive` property must be specified in the `Mina.Network` function to fetch actions from an Archive Node; otherwise, an error will occur.
```

Comment:
```suggestion
To configure SnarkyJS to use an Archive Node to fetch actions, use the `archive` property in the `Mina.Network` function. The `archive` property must be specified in the `Mina.Network` function to fetch actions from an Archive Node; otherwise, an error occurs.
```

Reply:
Building on Barrie's:

> If your smart contract needs to fetch actions from an archive node, simply provide an `archive` property within the object passed into `Mina.Network({mina: '...', archive: '...'})`. Its value should be the URL for the GraphQL API service that you want to use. If this property does not exist, an error will occur.

Future tense seems appropriate in this case (`an error will occur`) given it's an "if/then", no?